includes/controllers/class-user.php

```diff
@@ -1036,4 +1097,49 @@ public function render_user_view_page() {
 			]
 		) )->render();
 	}
+
+	/**
+	 * Redirects user view page.
+	 *
+	 * @return mixed
+	 */
+	public function redirect_user_resend_email_verify_page() {
+
+		if ( ! get_option( 'hp_user_verify_email' ) ) {
+			return true;
+		}
+
+		// Get user.
+		$user = Models\User::query()->get_by_id( absint( hivepress()->request->get_param( 'user_id' ) ) );
+
+		if ( ! $user ) {
+			wp_die( esc_html__( 'User not found', 'hivepress' ) );
```

Comment:
It's better to avoid wp_die, use it only for some critical errors (e.g. when the booking product is not generated), in case of already verified user you can simply redirect to the login/account page (so user_account route).